### PR TITLE
Add MythX smart contract security analysis

### PR DESCRIPTION
**Pull request by dmuhs**
### What was wrong / missing?
No security analysis reporting. This should provide a fix for #45.

### How was it fixed / added?
I integrated a new subcommand called `analyze` that uses the MythX API. The basic command usage is:
```
Brownie v1.0.0b12 - Python development framework for Ethereum

Usage: brownie analyze [options] [--async | --interval=<sec>]

Options:
  --gui                   Launch the Brownie GUI after analysis
  --full                  Perform a full scan (MythX Pro required)
  --interval=<sec>        Result polling interval in seconds [default: 3]
  --async                 Do not poll for results, print job IDs and exit
  --access-token          The JWT access token from the MythX dashboard
  --eth-address           The address of your MythX account
  --password              The password of your MythX account
  --help -h               Display this message

Use analyze to submit your project to the MythX API for smart contract
security analysis.

To authenticate with the MythX API, it is recommended that you provide
the MythX JWT access token. It can be obtained on the MythX dashboard
site in the profile section. They should be passed through the environment
variable "MYTHX_ACCESS_TOKEN". If that is not possible, it can also be
passed explicitly with the respective command line option.

Alternatively, you have to provide a username/password combination. It
is recommended to pass them through the environment variables as
"MYTHX_ETH_ADDRESS" and "MYTHX_PASSWORD".

You can also choose to not authenticate and submit your analyses as a free
trial user. No registration required! To see your past analyses, get access
to deeper vulnerability detection, and a neat dashboard, register at
https://mythx.io/. Any questions? Hit up dominik.muhs@consensys.net or contact
us on the website!
```

The plugin automatically generates a highlighting report in `reports/`. A simple example on the MetaCoin mix:

Screenshot from 2019-09-19 18-17-31

TODOs:
- [x] basic MVP (submit analyses, view reports)
- [x] support for full analyses
- [x] links to the MythX dashboard for submitted jobs (if logged in)
- [x] authentication by username/password and access JWT token
- [x] async mode to process long-running analyses without blocking
- [x] custom polling interval
- [x] basic tests
- [x] bump coverage even more
- [x] add documentation entries

### CODE REVIEW

**Comment by iamdefinitelyahuman** [comment]
Wow thanks for doing all this, it's super cool!  Been playing around with it, simple to execute and the results fit great with the GUI.

File: brownie/_cli/analyze.py [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+from brownie.exceptions import ProjectNotFound
+
+__doc__ = f"""Usage: brownie analyze [options] [--async | --interval=<sec>]
+
+Options:
+  --gui                   Launch the Brownie GUI after analysis
+  --full                  Perform a full scan (MythX Pro required)
+  --interval=<sec>        Result polling interval in seconds [default: 3]
+  --async                 Do not poll for results, print job IDs and exit
+  --access-token          The JWT access token from the MythX dashboard
```
Comment (iamdefinitelyahuman) [inline brownie/_cli/analyze.py]:
```suggestion
  --access-token <token>  The JWT access token from the MythX dashboard
```

need to include ``<token>`` or similar here for ``docopt`` to properly interpret the cli input.  Same for ``--eth-address`` and ``--password`` on the next lines.

File: brownie/_cli/analyze.py [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+    elif environ.get("MYTHX_ACCESS_TOKEN"):
+        authenticated = True
+        auth_args = {"access_token": environ.get("MYTHX_ACCESS_TOKEN")}
+    elif environ.get("MYTHX_ETH_ADDRESS") and environ.get("MYTHX_PASSWORD"):
+        authenticated = True
+        auth_args = {
+            "eth_address": environ.get("MYTHX_ETH_ADDRESS"),
+            "password": environ.get("MYTHX_PASSWORD"),
+        }
+    elif ARGV["eth-address"] and ARGV["password"]:
```
Comment (iamdefinitelyahuman) [inline brownie/_cli/analyze.py]: I would swap the positions of lines 108-110 and 102-107 so we have consistent behavior with cli arguments over-riding environment variables.

File: brownie/_cli/analyze.py (hunk 1 above)
Comment (dmuhs) [reply brownie/_cli/analyze.py]: Already fixed in 79b3556

File: tests/cli/test_analyze.py [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+    assert (
+        job_data["SafeMath"]["sources"]["contracts/SafeMath2.sol"]
+        == construct_source_dict_from_artifact(artifact)["contracts/SafeMath2.sol"]
+    )
+
+
+JOB_DATA = {
+    "SafeMath": {
+        "contract_name": "SafeMath",
+        "bytecode": "60556023600b82828239805160001a607314601657fe5b30600052607381538281f3fe73000000000000000000000000000000000000000030146080604052600080fdfea265627a7a72315820893edda58cfc8ab942d57dfc258288727bcaf844e6397a37aefea97a87eee32464736f6c634300050b0032",
```
Comment (iamdefinitelyahuman) [inline tests/cli/test_analyze.py]: could you add `# noqa E501` on this and the next few lines to it passes flake8? Then I think we can merge this!

**Pull request merged**